cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/sidebar_filtering.spec.js

```diff
@@ -0,0 +1,303 @@
+  }
+  // Check default column
+  getDocTableHeaderByIndex(1).should('have.text', '_source');
+  // Select all test fields
+  selectFields(testFields);
+  // Check that the default _source column no longer exists
+  getDocTableHeaderByIndex(1).should('not.have.text', '_source');
+  // Check table headers persistence between DQL and PPL
+  checkTableHeadersByArr(testFields);
+  if (indexPattern) {
+    dataExplorer.setQueryEditorLanguage('Lucene');
+    checkTableHeadersByArr(testFields);
+  }
+  dataExplorer.setQueryEditorLanguage('PPL');
+  checkTableHeadersByArr(testFields);
+  // Remove some fields
+  const firstTestField = testFields[0];
+  const secondTestField = testFields[1];
+  cy.getElementByTestId('fieldToggle-' + firstTestField).click();
+  cy.getElementByTestId('fieldToggle-' + secondTestField).click();
+  getDocTableHeaderByIndex(1).should('not.have.text', firstTestField);
+  getDocTableHeaderByIndex(2).should('not.have.text', secondTestField);
+  // Remove all fields
+  const thirdTestField = testFields[2];
+  const fourthTestField = testFields[3];
+  cy.getElementByTestId('fieldToggle-' + thirdTestField).click();
+  cy.getElementByTestId('fieldToggle-' + fourthTestField).click();
+  getDocTableHeaderByIndex(1).should('have.text', '_source');
+  getDocTableHeaderByIndex(2).should('not.exist');
+  // Select all test fields
+  selectFields(testFields);
+  // Check default column again
+  getDocTableHeaderByIndex(1).should('not.have.text', '_source');
+  // Check the columns match the selected fields
+  checkTableHeadersByArr(testFields);
+  if (indexPattern) {
+    // Validate default hits
+    cy.getElementByTestId('discoverQueryHits').should('have.text', '10,000');
+  }
+  // Send PPL query
+  cy.intercept('**/api/enhancements/search/ppl').as('pplQuery');
+  dataExplorer.sendQueryOnMultilineEditor(pplQuery);
+  cy.wait('@pplQuery').then(() => {
+    // Check table headers persistence after PPL query
+    cy.wait(1000);
+    checkTableHeadersByArr(testFields);
+    if (indexPattern) {
+      // Check filter was correctly applied
+      cy.getElementByTestId('discoverQueryHits').should('have.text', '1,152');
+    }
+    // Validate the first 5 rows on the _id column
+    checkDocTableColumnByArr(expectedValues, 2);
+  });
+  // Send SQL query
+  dataExplorer.setQueryEditorLanguage('OpenSearch SQL');
+  cy.intercept('**/api/enhancements/search/sql').as('sqlQuery');
+  dataExplorer.sendQueryOnMultilineEditor(sqlQuery);
+  cy.wait('@sqlQuery').then(() => {
+    // Check table headers persistence after SQL query
+    cy.wait(1000);
+    checkTableHeadersByArr(testFields);
+    // Validate the first 5 rows on the _id column
+    checkDocTableColumnByArr(expectedValues, 2);
+  });
+  // Clean all test fields for the next test
+  selectFields(testFields);
+};
+
+const checkFilteredFields = (indexPattern = true) => {
+  const filterFields = () => {
```

Comment:
This function seem to not be `filter(ing)Fields`, but rather verifying that the Sidebar filter bar data is correct. Could we update the name?